jsonschema-core.xml

```diff
@@ -570,13 +570,70 @@
                         As with the root schema, a subschema is either an object or a boolean.
                     </t>
                 </section>
+                <section title="Lexical Scope and Dynamic Scope">
+                    <t>
+                        While most JSON Schema keywords can be evaluated on their own,
+                        or at most need to take into account the values or results of
+                        adjacent keywords in the same schema object, a few have more
+                        complex behavior.
+                    </t>
+                    <t>
+                        The lexical scope of a keyword is deterimined by the nested JSON
+                        data structure of objects and arrays.  The largest such scope
+                        is an entire schema document.  The smallest scope is a single
+                        schema object with no subschemas.
+                    </t>
+                    <t>
+                        Keywords MAY be defined with a partial value, such as a URI-reference,
+                        which must be resolved against another value, such as another
+                        URI-reference or a full URI, which is found through the lexical
+                        structure of the JSON document.  The "$id" core keyword and
+                        the "base" JSON Hyper-Schema keyword are examples of this sort
+                        of behavior.  Additionally, "$ref" and "$recursiveRef" from
+                        this specification resolve their values in this way, although
+                        they do not change how further values are resolved.
```

Comment:
Thinking about this, I wonder if this is correct...
It IS correct, if we define that you must only have unique keys for each object, which I'm not sure we do, and is only a SHOULD according to the JSON specification.
If we already define such, please direct me to where.
If not, should we?
If not, I can explain a schema where this point would then be false!

Reply:
We explicitly exclude (or rather, defer behavior to the JSON parser that's being used) the duplicate property scenario here: https://tools.ietf.org/html/draft-handrews-json-schema-01#section-4.2.1